keyboards/preonic/keymaps/kjwon15/keymap.c

```diff
@@ -0,0 +1,374 @@
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          }
+          return false;
+          break;
+        case RAISE:
+          if (record->event.pressed) {
+            layer_on(_RAISE);
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          } else {
+            layer_off(_RAISE);
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          }
+          return false;
+          break;
+
+        // BACKLIT is not used. If you want to use, uncomment below
+        /*
+        case BACKLIT:
+          if (record->event.pressed) {
+            register_code(KC_RSFT);
+            #ifdef BACKLIGHT_ENABLE
+              backlight_step();
+            #endif
+            #ifdef __AVR__
+            PORTE &= ~(1<<6);
+            #endif
+          } else {
+            unregister_code(KC_RSFT);
+            #ifdef __AVR__
+            PORTE |= (1<<6);
+            #endif
+          }
+          return false;
+          break;
+        */
+
+        case KC_CESC:
+          if (record->event.pressed) {
+           cntl_interrupted = false;
+           cntl_timer = timer_read();
+           register_mods(MOD_BIT(KC_LCTL));
+          } else if (!cntl_interrupted && timer_elapsed(cntl_timer) < TAPPING_TERM) {
+           unregister_mods(MOD_BIT(KC_LCTL));
+           tap_code(KC_ESC);
+          } else {
+           unregister_mods(MOD_BIT(KC_LCTL));
+          }
+          return false;
+          break;
+
+        case SPC_MOU:
+          if (record->event.pressed) {
+           mouse_interrupted = false;
+           mouse_timer = timer_read();
+
+           cntl_interrupted = true;
+           layer_on(_MOUSE);
+          } else if (!mouse_interrupted && timer_elapsed(mouse_timer) < TAPPING_TERM) {
+           layer_off(_MOUSE);
+           tap_code(KC_SPC);
+          } else {
+           layer_off(_MOUSE);
+          }
+          return false;
+          break;
+
+        default:
+          cntl_interrupted = true;
+          mouse_interrupted = true;
+          break;
+      }
+    return true;
+};
+
+bool muse_mode = false;
+uint8_t last_muse_note = 0;
+uint16_t muse_counter = 0;
+uint8_t muse_offset = 70;
+uint16_t muse_tempo = 50;
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+  if (muse_mode) {
+    if (IS_LAYER_ON(_RAISE)) {
+      if (clockwise) {
+        muse_offset++;
+      } else {
+        muse_offset--;
+      }
+    } else {
+      if (clockwise) {
+        muse_tempo+=1;
+      } else {
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      tap_code(KC_PGDN);
+    } else {
+      tap_code(KC_PGUP);
+    }
+  }
+}
+
+void dip_update(uint8_t index, bool active) {
+  switch (index) {
+    case 0:
+      if (active) {
+        layer_on(_ADJUST);
+      } else {
+        layer_off(_ADJUST);
+      }
+      break;
+    case 1:
+      if (active) {
+        muse_mode = true;
+      } else {
+        muse_mode = false;
+        #ifdef AUDIO_ENABLE
```

Comment:
There are some changes to the default keymap that fix some issues with dip switch, and the startup sound.   You should take a look at those changes (to both this part, and to the matrix scan/muse stuff).

Reply:
Thank you!